### PR TITLE
Allocation Module

### PR DESCRIPTION
**Pull request by mvid**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -235,8 +235,10 @@ proto-check-breaking:
 
 TM_URL           = https://raw.githubusercontent.com/tendermint/tendermint/v0.34.0/proto/tendermint
 GOGO_PROTO_URL   = https://raw.githubusercontent.com/regen-network/protobuf/cosmos
-COSMOS_PROTO_URL = https://raw.githubusercontent.com/regen-network/cosmos-proto/main
+COSMOS_PROTO_URL = https://raw.githubusercontent.com/regen-network/cosmos-proto/master
 COSMOS_SDK_PROTO_URL = https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.40.0-rc6/proto/cosmos/base
```
Comment (jackzampolin) [inline Makefile:239]: Should we update this?

Comment (mvid) [reply Makefile]: Done

File: proto/allocation/v1/tx.proto [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+// MsgService defines the msgs that the oracle module handles.
+service Msg {
+     // DelegateAllocations defines a message that delegates the allocating to an account address.
+    rpc DelegateAllocations (MsgDelegateAllocations) returns (MsgDelegateAllocationsResponse);
+    // OracleDataPrevote defines a message that commits a hash of a oracle data feed before the data is actually submitted.
+    rpc AllocationPrecommit   (MsgAllocationPrecommit)   returns (MsgAllocationPrecommitResponse);
+    // OracleDataVote defines a message to submit the actual oracle data that was committed by the feeder through the prevote.
+    rpc AllocationCommit      (MsgAllocationCommit)      returns (MsgAllocationCommitResponse);
+}
```
Comment (jackzampolin) [inline proto/allocation/v1/tx.proto:18]: Do we still want precommit/reveal here if we are doing sparse voting?

File: proto/allocation/v1/allocation.proto [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+message TickWeights {
+  repeated TickWeight weights = 1;
+}
+
+
+////////////
+// Params //
+////////////
+
+// Cellar is a collection of pools for a token pair
```
Comment (jackzampolin) [inline proto/allocation/v1/allocation.proto:60]: Pending discussion with Zaki about data structures.

File: proto/allocation/v1/tx.proto (hunk 2 above)
Comment (mvid) [reply proto/allocation/v1/tx.proto:18]: Never heard the term "sparse voting" so if we aren't, it is news to me

**Comment by mvid** [comment]
@jackzampolin ready to merge

**Pull request merged**